test/Microsoft.AspNetCore.SystemWebAdapters.CoreServices.Tests/SessionState/Serialization/JsonSessionKeySerializerTests.cs

```diff
@@ -293,6 +294,70 @@ public void PrimitiveNullableSerializer(int? primitive)
         Assert.Equal(primitive, deserialized);
     }
 
+    [Fact]
+    public void CaseInsensitiveSessionKeys()
+    {
+        // Arrange
+        var key1 = _fixture.Create<string>();
+        var key2 = _fixture.Create<string>();
+
+        var options = new JsonSessionSerializerOptions
+        {
+            KeyComparer = StringComparer.OrdinalIgnoreCase,
+            KnownKeys =
+            {
+                { key1, typeof(Type1) },
+                { key2, typeof(int) },
+            }
+        };
+
+        // Act
+        var result1 = options.KnownKeys.ContainsKey(key1.ToUpperInvariant());
+        var result2 = options.KnownKeys.ContainsKey(key2);
+
+        // Assert
+        Assert.True(result1);
+        Assert.True(result2);
+    }
+
+    [Fact]
+    public void ChangeSessionKeysComparer()
```

Comment:
this test isn't really testing that the comparer is used, just that the keys are still there. Can you verify that the new comparer is use?

Reply:
Sure. It checks the key case insensitively though? The key (`key2`) upper case is verified it does not, and after the change that it does exists (it was registered lower case). That wouldn't work if it if the comparer didn't change?

How would you like me to check? Check the `Dictionary` itself?

```csharp
if (options.KnownKeys is Dictionary<string, Type> dict)
{
    Assert.Equal(dict.Comparer, StringComparer.OrdinalIgnoreCase);
}
```

Reply:
oh I guess I read the change too quickly. lgtm :)